scripts/prepare_int_data.py

```diff
@@ -0,0 +1,185 @@
+                                 "and to give it a standard format in preparation"
+                                 " for language model training.")
+
+parser.add_argument("--fold-dev-into", type=str,
+                    help="If supplied, the name of data-source into which to fold the "
+                    "counts of the dev data for purposes of vocabulary estimation "
+                    "(typically the same data source from which the dev data was "
+                    "originally excerpted).")
+parser.add_argument("--parallel", type=str, choices=["false", "true"], default="true",
+                    help="Setting --parallel false will disable the (default) "
+                    "parallel processing of multiple data sources by this script.")
+parser.add_argument("text_dir", 
+                    help="Directory of original data")
+parser.add_argument("vocab", 
+                    help="Vocabulary")
+parser.add_argument("int_dir", 
+                    help="Directory of formed data")
+
+args = parser.parse_args()
+
+# Add the script dir and the src dir to the path.
+os.environ['PATH'] = (os.environ['PATH'] + os.pathsep +
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + os.pathsep +
+                      os.path.abspath(os.path.dirname(sys.argv[0])) + "/../src");
+
+# do some checking
+if os.system("validate_text_dir.py " + args.text_dir) != 0:
+  sys.exit("prepare_int_data.py: failed to validate text directory")
+
+if os.system("validate_vocab.py " + args.vocab) != 0:
+  sys.exit("prepare_int_data.py: failed to validate vocab")
+
+if not os.path.exists(os.path.abspath(os.path.dirname(sys.argv[0])) + "/text_to_int.py"):
+  sys.exit("prepare_int_data.py: expected text_to_int.py to be on the path")
+
+# create the output data directory
+if not os.path.exists(args.int_dir):
+  os.makedirs(args.int_dir)
+
+if not os.path.exists(args.int_dir + "/log"):
+  os.makedirs(args.int_dir + "/log")
+
+# remove any old *.int.gz files in the output data directory
+filelist = [f for f in os.listdir(args.int_dir) if f.endswith(".int.gz")]
+for f in filelist:
+  try:
+    os.remove(f)
+  except OSError:
+    pass
+
+def ExitProgram(message):
+  print(message, file=sys.stderr)
+  os._exit(1)
+
+def RunCommand(command):
+  # print the command for logging
+  print(command, file=sys.stderr)
+  if os.system(command) != 0:
+    ExitProgram("prepare_int_data.py: error running command: " + command)
+
+def GetCommandStdout(command):
+  # print the command for logging
+  print(command, file=sys.stderr)
+  try:
+    output = subprocess.check_output(command, shell = True)
+  except:
+    ExitProgram("prepare_int_data.py: error running command: " + command)
+  return output
+
+def GetNames(script_dir, text_dir, int_dir):
+  command = "{0}/internal/get_names.py {1} > {2}/names".format(
+              script_dir, text_dir, int_dir) 
+  RunCommand(command)
+
+def GetNumTrainSets(int_dir):
+  command = "cat {0}/names | wc -l".format(int_dir)
```

Comment:
I don't think using 'cat' and 'ls'  is a very Pythonic way to do this, would be better to iterate over lines using readline() or readlines().
But regardless, it would be easier to get this number from the file 'num_train_sets', which will exist in the input directory.
See how other scripts read num_train_sets.